golint/golint.go

```diff
@@ -69,25 +73,61 @@ func exists(filename string) bool {
 }
 
 func lintFiles(filenames ...string) {
-	files := make(map[string][]byte)
+	files := make(map[string]struct{})
+	pkgs := make(map[string]map[string][]byte)
 	for _, filename := range filenames {
+		if _, ok := files[filename]; ok {
+			continue
+		}
+
 		src, err := ioutil.ReadFile(filename)
 		if err != nil {
 			fmt.Fprintln(os.Stderr, err)
 			continue
 		}
-		files[filename] = src
+		files[filename] = struct{}{}
+
+		pkgName := path.Dir(filename)
+		if strings.HasSuffix(filename, "_test.go") {
```

Comment:
Oh man! That was some wasted time... I totally forgot that go/loader does not expose the raw data of the loaded files and it is needed to display the problems. If you are interested https://github.com/zimmski/lint/commit/2c0374a88058e315aab1eaa036ea17bbe21181f6
The main could look really cleaned up if some features where added to go/loader:
- Expose raw content of loaded files
- Allow to load directories
- Allow CreateFromFilename to load from different packages
- Allow to disable the printing of errors if a loaded file has unknown types and such
- Pretty much the whole main could be done in Config.FromArgs 
Maybe I should join the go/loader fun party to make these things possible. A LOT of Go tools could benefit from that.

@dsymonds I mention you just to be sure. Please revisit this PR and comments on the go/loader attempt would be cool too.